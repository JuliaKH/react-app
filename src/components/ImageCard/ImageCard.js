import React, {Component} from 'react';
import './ImageCard.scss'

class ImageCard extends Component {
    constructor(props) {
        super(props);
        // this.state = { spans: 0 };
        // this.imageRef = React.createRef();
        // this.state = {image: {}};
        this.image = {};
    }

    componentDidMount() {
        // this.imageRef.current.addEventListener('load', this.setSpans);
        // this.setState({ image: this.getImage() });
        // console.log(this.image);
    }
    getImage = () => {
        const images = this.props.images;
        let image = images.filter(img =>
            img.id === this.props.match.params.id
        );
        return image[0];
    };
    // setSpans = () => {
    //     const height = this.imageRef.current.clientHeight;
    //     const spansRows = Math.ceil(height / 150 );
    //     this.setState({ spans: spansRows });
    // };)
    render() {
        this.image = this.getImage();
        return (
            <div className="card">
            {/*<div style={{gridRowEnd: `span ${this.state.spans}`}}>*/}
            {/*    <img ref={this.imageRef} src={this.props.image.urls.regular}*/}
            {/*         alt={this.props.image.alt_description} className="images__list-img" />*/}
                <img src={this.image.urls.regular}
                     alt={this.image.alt_description} className="card__image" />
            </div>
        );
    }
}

export default ImageCard;
