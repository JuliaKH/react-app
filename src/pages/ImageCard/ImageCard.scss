.card{
  max-width: 800px;
  border: 1px solid #c0c0c0;
  box-shadow: 0 0 10px rgba(0,0,0,0.5);
  padding: 50px;
  margin: 30px auto 30px auto;
  &__image{
    max-width: 500px;
    width: 100%;
    margin: 0 auto;
    display: block;
  }
}

