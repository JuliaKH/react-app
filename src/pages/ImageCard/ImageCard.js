import React, {Component} from 'react';
import './ImageCard.scss'
import axios from "axios";

class ImageCard extends Component {
    constructor(props){
        super(props);
        this.state = {image: {
            urls: {}
            }}
    }

    componentDidMount() {
        this.getImageById(this.props.match.params.id);
    }

    getImageById = (id) => {
        const options = {
            params: {
            },
            headers: {
                Authorization: 'Client-ID 5110e0875d03049c42ef2483cf9a9ad53c6a0f46dd526e9ee18dca0c3c6a8f0b'
            }
        };
        axios
            .get(`https://api.unsplash.com/photos/${id}`, options)
            .then(res => {
                this.setState({image: res.data});
            })
            .catch(err => {
            });
    };
    render() {
        console.log(this.state.image.urls);
        return (
            <section className="card">
                <img src={this.state.image.urls.regular}  alt={this.state.image.alt_description} className="card__image" />
            </section>
        );
    }
}
ImageCard.defaultProps = {
    match: { params: { code: 123 } },
};
export default ImageCard;
