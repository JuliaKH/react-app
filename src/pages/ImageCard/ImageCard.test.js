import React from 'react';
import ImageCard  from "./ImageCard";
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import {BrowserRouter} from "react-router-dom";
import axios from 'axios';

describe('ImageCard ', () => {

    module.exports = {
        get: jest.fn(() => {
            return Promise.resolve({
                data: [
                    {
                        urls: {
                            regular: 'some.jpg'
                        }
                    }
                ]
            })
        })
    };
    jest.mock('axios');


    it('should render correctly', () => {
        const wrapper = shallow(<ImageCard />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should fetch a image', () => {
        const getSpy = jest.spyOn(axios, 'get');
        const wrapper = shallow(<ImageCard />);
        expect(getSpy).toBeCalled()
    });

    it('should call componentDidMount', () => {
        jest.spyOn(ImageCard.prototype, 'componentDidMount');
        const wrapper = shallow(<ImageCard />);
        expect(ImageCard.prototype.componentDidMount.mock.calls.length).toBe(1);
    });

    // it('should set image to state', () => {
    //     const getSpy = jest.spyOn(axios, 'get');
    //     const wrapper = shallow(<ImageCard />);
    //     jest.runAllTimers();
    //     // component.update();
    //     expect(wrapper.state().image.urls).toEqual(true);
    //
    // });

});
