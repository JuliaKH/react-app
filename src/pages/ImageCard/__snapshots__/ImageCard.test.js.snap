// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard  should render correctly 1`] = `
<section
  className="card"
>
  <img
    className="card__image"
  />
</section>
`;
